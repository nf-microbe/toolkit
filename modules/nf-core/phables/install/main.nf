process PHABLES_INSTALL {
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://carsonjm/phables:1.4.0':
        'docker.io/carsonjm/phables:1.4.0' }"

    output:
    path "phables_db"   , emit: phables_db
    path "versions.yml" , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    mkdir phables_db

    wget -P phables_db https://raw.githubusercontent.com/metagentools/MetaCoAG/develop/src/metacoag/metacoag_utils/auxiliary/marker.hmm
    wget -P phables_db https://phrogs.lmge.uca.fr/downloads_from_website/phrog_annot_v4.tsv

    wget -P phables_db https://phrogs.lmge.uca.fr/downloads_from_website/phrogs_mmseqs_db.tar.gz
    tar -xvf phables_db/phrogs_mmseqs_db.tar.gz -C phables_db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phables: \$(echo \$(phables -v 2>&1) | sed -n 's/phables, version //' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    """
    mkdir phables_db
    touch phables_db/marker.hmm
    touch phables_db/phrog_annot_v4.tsv
    mkdir phables_db/phrogs_mmseqs_db
    touch phables_db/phrogs_mmseqs_db/phrogs_db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phables: \$(echo \$(phables -v 2>&1) | sed -n 's/phables, version //' ))
    END_VERSIONS
    """
}
