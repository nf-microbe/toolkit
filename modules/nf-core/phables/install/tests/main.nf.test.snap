{
    "No input required": {
        "content": [
            {
                "0": [
                    [
                        "marker.hmm:md5,3b63057612a5512871dd3a0e158af824",
                        "phrog_annot_v4.tsv:md5,3f078c589efeef3352c5e60c1ced9a44",
                        [
                            "phrogs_db:md5,75649b7e711af909288b94ebaec8e724",
                            "phrogs_db.dbtype:md5,c31b034959815c44d4dafd6f30846159",
                            "phrogs_db.index:md5,cd1b2517d80797c8e2fb0f6666482a1d",
                            "phrogs_profile_db:md5,d355c1b2599c9fa1a2e19b587cc3e75a",
                            "phrogs_profile_db.dbtype:md5,f2dd0dedb2c260419ece4a9e03b2e828",
                            "phrogs_profile_db.index:md5,61d1b0e42f135379b553c8f48372ca44",
                            "phrogs_profile_db_consensus:md5,4107609c2701b2eca2f561177d1498f3",
                            "phrogs_profile_db_consensus.dbtype:md5,f1d3ff8443297732862df21dc4e57262",
                            "phrogs_profile_db_consensus.index:md5,a711c8e652b63aa2aceb20716ae8c600",
                            "phrogs_profile_db_h:md5,25c9cbd7ac212a198f6da1e1883ea5b9",
                            "phrogs_profile_db_h.index:md5,ae5e93f21c408190ea41e1bbf1b10ff4",
                            "phrogs_profile_db_seq:md5,566a77a95040aeed03d72f2d7a369607",
                            "phrogs_profile_db_seq.dbtype:md5,f1d3ff8443297732862df21dc4e57262",
                            "phrogs_profile_db_seq.index:md5,a711c8e652b63aa2aceb20716ae8c600",
                            "phrogs_profile_db_seq_h:md5,25c9cbd7ac212a198f6da1e1883ea5b9",
                            "phrogs_profile_db_seq_h.index:md5,ae5e93f21c408190ea41e1bbf1b10ff4"
                        ],
                        "phrogs_mmseqs_db.tar.gz:md5,7cdeee285789a73ad836829f13311669"
                    ]
                ],
                "1": [
                    "versions.yml:md5,de9904ebb8e74d9a998661982d83dfd0"
                ],
                "phables_db": [
                    [
                        "marker.hmm:md5,3b63057612a5512871dd3a0e158af824",
                        "phrog_annot_v4.tsv:md5,3f078c589efeef3352c5e60c1ced9a44",
                        [
                            "phrogs_db:md5,75649b7e711af909288b94ebaec8e724",
                            "phrogs_db.dbtype:md5,c31b034959815c44d4dafd6f30846159",
                            "phrogs_db.index:md5,cd1b2517d80797c8e2fb0f6666482a1d",
                            "phrogs_profile_db:md5,d355c1b2599c9fa1a2e19b587cc3e75a",
                            "phrogs_profile_db.dbtype:md5,f2dd0dedb2c260419ece4a9e03b2e828",
                            "phrogs_profile_db.index:md5,61d1b0e42f135379b553c8f48372ca44",
                            "phrogs_profile_db_consensus:md5,4107609c2701b2eca2f561177d1498f3",
                            "phrogs_profile_db_consensus.dbtype:md5,f1d3ff8443297732862df21dc4e57262",
                            "phrogs_profile_db_consensus.index:md5,a711c8e652b63aa2aceb20716ae8c600",
                            "phrogs_profile_db_h:md5,25c9cbd7ac212a198f6da1e1883ea5b9",
                            "phrogs_profile_db_h.index:md5,ae5e93f21c408190ea41e1bbf1b10ff4",
                            "phrogs_profile_db_seq:md5,566a77a95040aeed03d72f2d7a369607",
                            "phrogs_profile_db_seq.dbtype:md5,f1d3ff8443297732862df21dc4e57262",
                            "phrogs_profile_db_seq.index:md5,a711c8e652b63aa2aceb20716ae8c600",
                            "phrogs_profile_db_seq_h:md5,25c9cbd7ac212a198f6da1e1883ea5b9",
                            "phrogs_profile_db_seq_h.index:md5,ae5e93f21c408190ea41e1bbf1b10ff4"
                        ],
                        "phrogs_mmseqs_db.tar.gz:md5,7cdeee285789a73ad836829f13311669"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de9904ebb8e74d9a998661982d83dfd0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-26T06:01:50.812000414"
    },
    "No input required - stub": {
        "content": [
            {
                "0": [
                    [
                        "marker.hmm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "phrog_annot_v4.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        [
                            "phrogs_db:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,de9904ebb8e74d9a998661982d83dfd0"
                ],
                "phables_db": [
                    [
                        "marker.hmm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "phrog_annot_v4.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        [
                            "phrogs_db:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de9904ebb8e74d9a998661982d83dfd0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-26T06:02:22.645809715"
    }
}